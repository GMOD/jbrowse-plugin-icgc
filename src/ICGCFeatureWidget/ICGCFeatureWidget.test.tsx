import React from 'react'
import { types } from 'mobx-state-tree'
import { render } from '@testing-library/react'
import { ConfigurationSchema } from '@jbrowse/core/configuration'
import ICGCFeatureDetails from './ICGCFeatureWidget'

describe('ICGCTrack widget', () => {
  it('renders a mutation with the required model elements', () => {
    const f = types
      .model({
        // used as proxy for session
        pluginManager: types.optional(types.frozen(), {}),
        rpcManager: types.optional(types.frozen(), {}),
        // used as proxy for session
        configuration: ConfigurationSchema('null', {}),
        featureData: types.frozen({
          refName: 'chr3',
          type: 'single base substitution',
          start: 124907170,
          end: 124907171,
          id: 'MU125557849',
          chromosome: 3,
          mutation: 'G>T',
          assemblyVersion: 'GRCh37',
          referenceGenomeAllele: 'G',
          testedDonorCount: 186,
          affectedDonorCountTotal: 1,
          affectedDonorCountFiltered: 1,
          affectedProjectCount: 1,
          functionalImpact: 'Unknown',
        }),
      })
      .create()

    const { container } = render(<ICGCFeatureDetails model={f} />)
    expect(container.firstChild).toMatchSnapshot()
  })

  it('renders a gene with the required model elements', () => {
    const f = types
      .model({
        // used as proxy for session
        rpcManager: 'hello',
        // used as proxy for session
        configuration: ConfigurationSchema('null', {}),
        featureData: types.frozen({
          refName: 'chr3',
          type: 'MU128994531',
          start: 124909350,
          end: 124909349,
          id: 'DO52170',
          donorId: 'DO52170',
          mutationId: 'MU128994531',
          chromosome: 3,
          projectId: 'LIHC-US',
          mutation: 'G>T',
        }),
      })
      .create()

    const { container } = render(<ICGCFeatureDetails model={f} />)
    expect(container.firstChild).toMatchSnapshot()
  })
})
