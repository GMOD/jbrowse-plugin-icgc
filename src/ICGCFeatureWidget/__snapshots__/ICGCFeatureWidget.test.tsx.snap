// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ICGCTrack widget renders a gene with the required model elements 1`] = `
<div
  class="MuiPaper-root MuiPaper-elevation1 MuiPaper-rounded"
  data-testid="icgc-widget"
>
  <div
    class="MuiPaper-root MuiAccordion-root Mui-expanded MuiAccordion-rounded MuiPaper-elevation1 MuiPaper-rounded"
  >
    <div
      aria-disabled="false"
      aria-expanded="true"
      class="MuiButtonBase-root MuiAccordionSummary-root Mui-expanded"
      role="button"
      tabindex="0"
    >
      <div
        class="MuiAccordionSummary-content Mui-expanded"
      >
        <span
          class="MuiTypography-root MuiTypography-button"
        >
           
          DO52170 - MU128994531
        </span>
      </div>
      <div
        aria-disabled="false"
        aria-hidden="true"
        class="MuiButtonBase-root MuiIconButton-root MuiAccordionSummary-expandIcon Mui-expanded MuiIconButton-edgeEnd"
      >
        <span
          class="MuiIconButton-label"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root makeStyles-expandIcon-12"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M16.59 8.59L12 13.17 7.41 8.59 6 10l6 6 6-6z"
            />
          </svg>
        </span>
        <span
          class="MuiTouchRipple-root"
        />
      </div>
    </div>
    <div
      class="MuiCollapse-root MuiCollapse-entered"
      style="min-height: 0px;"
    >
      <div
        class="MuiCollapse-wrapper"
      >
        <div
          class="MuiCollapse-wrapperInner"
        >
          <div
            role="region"
          >
            <div
              class="MuiAccordionDetails-root makeStyles-expansionPanelDetails-11"
            >
              <p
                class="MuiTypography-root MuiTypography-body1"
              >
                Core details
              </p>
              <div
                class="makeStyles-field-13"
              >
                <div
                  class="makeStyles-fieldName-15"
                >
                  Position
                </div>
                <div
                  class="makeStyles-fieldValue-16"
                >
                  <div>
                    chr3:124,909,351..124,909,349 
                  </div>
                </div>
              </div>
              <div
                class="makeStyles-field-13"
              >
                <div
                  class="makeStyles-fieldName-15"
                >
                  Length
                </div>
                <div
                  class="makeStyles-fieldValue-16"
                >
                  <div>
                    -1
                  </div>
                </div>
              </div>
              <div
                class="makeStyles-field-13"
              >
                <div
                  class="makeStyles-fieldName-15"
                >
                  Type
                </div>
                <div
                  class="makeStyles-fieldValue-16"
                >
                  <div>
                    MU128994531
                  </div>
                </div>
              </div>
              <hr
                class="MuiDivider-root"
              />
              <p
                class="MuiTypography-root MuiTypography-body1"
              >
                Attributes
              </p>
              <div
                class="makeStyles-field-13"
              >
                <div
                  class="makeStyles-fieldName-15"
                >
                  id
                </div>
                <div
                  class="makeStyles-fieldValue-16"
                >
                  <div>
                    DO52170
                  </div>
                </div>
              </div>
              <div
                class="makeStyles-field-13"
              >
                <div
                  class="makeStyles-fieldName-15"
                >
                  donorId
                </div>
                <div
                  class="makeStyles-fieldValue-16"
                >
                  <div>
                    DO52170
                  </div>
                </div>
              </div>
              <div
                class="makeStyles-field-13"
              >
                <div
                  class="makeStyles-fieldName-15"
                >
                  mutationId
                </div>
                <div
                  class="makeStyles-fieldValue-16"
                >
                  <div>
                    MU128994531
                  </div>
                </div>
              </div>
              <div
                class="makeStyles-field-13"
              >
                <div
                  class="makeStyles-fieldName-15"
                >
                  chromosome
                </div>
                <div
                  class="makeStyles-fieldValue-16"
                >
                  <div>
                    3
                  </div>
                </div>
              </div>
              <div
                class="makeStyles-field-13"
              >
                <div
                  class="makeStyles-fieldName-15"
                >
                  projectId
                </div>
                <div
                  class="makeStyles-fieldValue-16"
                >
                  <div>
                    LIHC-US
                  </div>
                </div>
              </div>
              <div
                class="makeStyles-field-13"
              >
                <div
                  class="makeStyles-fieldName-15"
                >
                  mutation
                </div>
                <div
                  class="makeStyles-fieldValue-16"
                >
                  <div>
                    G&gt;T
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    class="MuiPaper-root MuiAccordion-root Mui-expanded MuiAccordion-rounded MuiPaper-elevation1 MuiPaper-rounded"
  >
    <div
      aria-disabled="false"
      aria-expanded="true"
      class="MuiButtonBase-root MuiAccordionSummary-root Mui-expanded"
      role="button"
      tabindex="0"
    >
      <div
        class="MuiAccordionSummary-content Mui-expanded"
      >
        <span
          class="MuiTypography-root MuiTypography-button"
        >
           
          External Links
        </span>
      </div>
      <div
        aria-disabled="false"
        aria-hidden="true"
        class="MuiButtonBase-root MuiIconButton-root MuiAccordionSummary-expandIcon Mui-expanded MuiIconButton-edgeEnd"
      >
        <span
          class="MuiIconButton-label"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root makeStyles-expandIcon-12"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M16.59 8.59L12 13.17 7.41 8.59 6 10l6 6 6-6z"
            />
          </svg>
        </span>
        <span
          class="MuiTouchRipple-root"
        />
      </div>
    </div>
    <div
      class="MuiCollapse-root MuiCollapse-entered"
      style="min-height: 0px;"
    >
      <div
        class="MuiCollapse-wrapper"
      >
        <div
          class="MuiCollapse-wrapperInner"
        >
          <div
            role="region"
          >
            <div
              class="MuiAccordionDetails-root makeStyles-expansionPanelDetails-11"
            >
              <div
                class="makeStyles-innerCard-20"
              >
                <table
                  class="MuiTable-root makeStyles-table-18"
                >
                  <tbody
                    class="MuiTableBody-root"
                  >
                    <tr
                      class="MuiTableRow-root"
                    >
                      <td
                        class="MuiTableCell-root MuiTableCell-body"
                      >
                        ICGC Data Portal
                      </td>
                      <td
                        class="MuiTableCell-root MuiTableCell-body"
                      >
                        <a
                          class="MuiTypography-root MuiLink-root MuiLink-underlineAlways makeStyles-link-19 MuiTypography-colorPrimary"
                          href="https://dcc.icgc.org/mutations/MU128994531"
                          rel="noopener"
                          target="_blank"
                        >
                          MU128994531
                        </a>
                      </td>
                    </tr>
                  </tbody>
                </table>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`ICGCTrack widget renders a mutation with the required model elements 1`] = `
<div
  class="MuiPaper-root MuiPaper-elevation1 MuiPaper-rounded"
  data-testid="icgc-widget"
>
  <div
    class="MuiPaper-root MuiAccordion-root Mui-expanded MuiAccordion-rounded MuiPaper-elevation1 MuiPaper-rounded"
  >
    <div
      aria-disabled="false"
      aria-expanded="true"
      class="MuiButtonBase-root MuiAccordionSummary-root Mui-expanded"
      role="button"
      tabindex="0"
    >
      <div
        class="MuiAccordionSummary-content Mui-expanded"
      >
        <span
          class="MuiTypography-root MuiTypography-button"
        >
           
          MU125557849 - single base substitution
        </span>
      </div>
      <div
        aria-disabled="false"
        aria-hidden="true"
        class="MuiButtonBase-root MuiIconButton-root MuiAccordionSummary-expandIcon Mui-expanded MuiIconButton-edgeEnd"
      >
        <span
          class="MuiIconButton-label"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root makeStyles-expandIcon-2"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M16.59 8.59L12 13.17 7.41 8.59 6 10l6 6 6-6z"
            />
          </svg>
        </span>
        <span
          class="MuiTouchRipple-root"
        />
      </div>
    </div>
    <div
      class="MuiCollapse-root MuiCollapse-entered"
      style="min-height: 0px;"
    >
      <div
        class="MuiCollapse-wrapper"
      >
        <div
          class="MuiCollapse-wrapperInner"
        >
          <div
            role="region"
          >
            <div
              class="MuiAccordionDetails-root makeStyles-expansionPanelDetails-1"
            >
              <p
                class="MuiTypography-root MuiTypography-body1"
              >
                Core details
              </p>
              <div
                class="makeStyles-field-3"
              >
                <div
                  class="makeStyles-fieldName-5"
                >
                  Position
                </div>
                <div
                  class="makeStyles-fieldValue-6"
                >
                  <div>
                    chr3:124,907,171..124,907,171 
                  </div>
                </div>
              </div>
              <div
                class="makeStyles-field-3"
              >
                <div
                  class="makeStyles-fieldName-5"
                >
                  Length
                </div>
                <div
                  class="makeStyles-fieldValue-6"
                >
                  <div>
                    1
                  </div>
                </div>
              </div>
              <div
                class="makeStyles-field-3"
              >
                <div
                  class="makeStyles-fieldName-5"
                >
                  Type
                </div>
                <div
                  class="makeStyles-fieldValue-6"
                >
                  <div>
                    single base substitution
                  </div>
                </div>
              </div>
              <hr
                class="MuiDivider-root"
              />
              <p
                class="MuiTypography-root MuiTypography-body1"
              >
                Attributes
              </p>
              <div
                class="makeStyles-field-3"
              >
                <div
                  class="makeStyles-fieldName-5"
                >
                  id
                </div>
                <div
                  class="makeStyles-fieldValue-6"
                >
                  <div>
                    MU125557849
                  </div>
                </div>
              </div>
              <div
                class="makeStyles-field-3"
              >
                <div
                  class="makeStyles-fieldName-5"
                >
                  chromosome
                </div>
                <div
                  class="makeStyles-fieldValue-6"
                >
                  <div>
                    3
                  </div>
                </div>
              </div>
              <div
                class="makeStyles-field-3"
              >
                <div
                  class="makeStyles-fieldName-5"
                >
                  mutation
                </div>
                <div
                  class="makeStyles-fieldValue-6"
                >
                  <div>
                    G&gt;T
                  </div>
                </div>
              </div>
              <div
                class="makeStyles-field-3"
              >
                <div
                  class="makeStyles-fieldName-5"
                >
                  assemblyVersion
                </div>
                <div
                  class="makeStyles-fieldValue-6"
                >
                  <div>
                    GRCh37
                  </div>
                </div>
              </div>
              <div
                class="makeStyles-field-3"
              >
                <div
                  class="makeStyles-fieldName-5"
                >
                  referenceGenomeAllele
                </div>
                <div
                  class="makeStyles-fieldValue-6"
                >
                  <div>
                    G
                  </div>
                </div>
              </div>
              <div
                class="makeStyles-field-3"
              >
                <div
                  class="makeStyles-fieldName-5"
                >
                  testedDonorCount
                </div>
                <div
                  class="makeStyles-fieldValue-6"
                >
                  <div>
                    186
                  </div>
                </div>
              </div>
              <div
                class="makeStyles-field-3"
              >
                <div
                  class="makeStyles-fieldName-5"
                >
                  affectedDonorCountTotal
                </div>
                <div
                  class="makeStyles-fieldValue-6"
                >
                  <div>
                    1
                  </div>
                </div>
              </div>
              <div
                class="makeStyles-field-3"
              >
                <div
                  class="makeStyles-fieldName-5"
                >
                  affectedDonorCountFiltered
                </div>
                <div
                  class="makeStyles-fieldValue-6"
                >
                  <div>
                    1
                  </div>
                </div>
              </div>
              <div
                class="makeStyles-field-3"
              >
                <div
                  class="makeStyles-fieldName-5"
                >
                  affectedProjectCount
                </div>
                <div
                  class="makeStyles-fieldValue-6"
                >
                  <div>
                    1
                  </div>
                </div>
              </div>
              <div
                class="makeStyles-field-3"
              >
                <div
                  class="makeStyles-fieldName-5"
                >
                  functionalImpact
                </div>
                <div
                  class="makeStyles-fieldValue-6"
                >
                  <div>
                    Unknown
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    class="MuiPaper-root MuiAccordion-root Mui-expanded MuiAccordion-rounded MuiPaper-elevation1 MuiPaper-rounded"
  >
    <div
      aria-disabled="false"
      aria-expanded="true"
      class="MuiButtonBase-root MuiAccordionSummary-root Mui-expanded"
      role="button"
      tabindex="0"
    >
      <div
        class="MuiAccordionSummary-content Mui-expanded"
      >
        <span
          class="MuiTypography-root MuiTypography-button"
        >
           
          External Links
        </span>
      </div>
      <div
        aria-disabled="false"
        aria-hidden="true"
        class="MuiButtonBase-root MuiIconButton-root MuiAccordionSummary-expandIcon Mui-expanded MuiIconButton-edgeEnd"
      >
        <span
          class="MuiIconButton-label"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root makeStyles-expandIcon-2"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M16.59 8.59L12 13.17 7.41 8.59 6 10l6 6 6-6z"
            />
          </svg>
        </span>
        <span
          class="MuiTouchRipple-root"
        />
      </div>
    </div>
    <div
      class="MuiCollapse-root MuiCollapse-entered"
      style="min-height: 0px;"
    >
      <div
        class="MuiCollapse-wrapper"
      >
        <div
          class="MuiCollapse-wrapperInner"
        >
          <div
            role="region"
          >
            <div
              class="MuiAccordionDetails-root makeStyles-expansionPanelDetails-1"
            >
              <div
                class="makeStyles-innerCard-10"
              >
                <table
                  class="MuiTable-root makeStyles-table-8"
                >
                  <tbody
                    class="MuiTableBody-root"
                  >
                    <tr
                      class="MuiTableRow-root"
                    >
                      <td
                        class="MuiTableCell-root MuiTableCell-body"
                      >
                        ICGC Data Portal
                      </td>
                      <td
                        class="MuiTableCell-root MuiTableCell-body"
                      >
                        <a
                          class="MuiTypography-root MuiLink-root MuiLink-underlineAlways makeStyles-link-9 MuiTypography-colorPrimary"
                          href="https://dcc.icgc.org/mutations/MU125557849"
                          rel="noopener"
                          target="_blank"
                        >
                          MU125557849
                        </a>
                      </td>
                    </tr>
                  </tbody>
                </table>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
